Code/Tools/ProjectManager/Source/ProjectExportController.cpp

```diff
@@ -0,0 +1,146 @@
+        SettingsInterface::Get()->SetProjectBuiltSuccessfully(m_projectInfo, false);
+
+        connect(&m_workerThread, &QThread::finished, m_worker, &ProjectExportWorker::deleteLater);
+        connect(&m_workerThread, &QThread::started, m_worker, &ProjectExportWorker::ExportProject);
+        connect(m_worker, &ProjectExportWorker::Done, this, &ProjectExportController::HandleResults);
+        connect(m_worker, &ProjectExportWorker::UpdateProgress, this, &ProjectExportController::UpdateUIProgress);
+    }
+
+    ProjectExportController::~ProjectExportController()
+    {
+        m_workerThread.requestInterruption();
+        m_workerThread.quit();
+        m_workerThread.wait();
+    }
+
+    void ProjectExportController::Start()
+    {
+        m_workerThread.start();
+        UpdateUIProgress("");
+    }
+
+    void ProjectExportController::SetProjectButton(ProjectButton* projectButton)
+    {
+        m_projectButton = projectButton;
+
+        if (projectButton)
+        {
+            projectButton->SetProjectButtonAction(tr("Cancel"), [this] { HandleCancel(); });
+            AZ::Outcome<QString, QString> logFilePathQuery = m_worker->GetLogFilePath();
+            if (logFilePathQuery.IsSuccess())
+            {
+                projectButton->SetBuildLogsLink(QUrl::fromLocalFile(logFilePathQuery.GetValue()));
+            }
+            projectButton->SetState(ProjectButtonState::Exporting);
+
+            if (!m_lastLine.isEmpty())
+            {
+                UpdateUIProgress(m_lastLine);
+            }
+        }
+    }
+
+    const ProjectInfo& ProjectExportController::GetProjectInfo() const
+    {
+        return m_projectInfo;
+    }
+
+    void ProjectExportController::UpdateUIProgress(const QString& lastLine)
+    {
+        m_lastLine = lastLine.left(s_maxDisplayedBuiltOutputChars);
+
+        if (m_projectButton)
+        {
+            m_projectButton->SetContextualText(m_lastLine);
+        }
+    }
+
+    void ProjectExportController::HandleResults(const QString& result)
+    {
+        if (!result.isEmpty())
+        {
+            AZ::Outcome<QString, QString> logFilePathQuery = m_worker->GetLogFilePath();
+            if (result.contains(tr("log")))
+            {
+                QMessageBox::StandardButton openLog = QMessageBox::critical(
+                    m_parent,
+                    tr(LauncherExportFailedMessage),
+                    result + tr("\n\nWould you like to view log?"),
+                    QMessageBox::No | QMessageBox::Yes);
+
+                if (openLog == QMessageBox::Yes)
+                {
+                    // Open application assigned to this file type
+                    if (!logFilePathQuery.IsSuccess() || !QDesktopServices::openUrl(QUrl::fromLocalFile(logFilePathQuery.GetValue())))
+                    {
+                        qDebug() << "QDesktopServices::openUrl failed to open " << m_projectInfo.m_logUrl.toString() << "\n";
```

Comment:
If `!logFilePathQuery.IsSuccess()` condition is true, then the debug statement `qDebug() << "QDesktopServices::openUrl failed to open " << m_projectInfo.m_logUrl.toString() << "\n";` is not correct. It will short-circuit and skip the `openUrl` command therefore making that error misleading. It needs to instead show the actual error from logFilePathQuery.